src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -257,10 +257,24 @@ namespace OpenMS
 
   void MRMIonSeries::annotateTransition(ReactionMonitoringTransition& tr, const TargetedExperiment::Peptide peptide, const double precursor_mz_threshold, double product_mz_threshold, bool enable_reannotation, std::vector<String> fragment_types, std::vector<size_t> fragment_charges, bool enable_specific_losses, bool enable_unspecific_losses, int round_decPow)
   {
-    CVTermList interpretation;
+    OPENMS_PRECONDITION(peptide.hasCharge(), "Cannot annotate transition without a peptide charge state")
+    // TODO: we should not have transitions without charge states
+    // OPENMS_PRECONDITION(tr.isProductChargeStateSet(), "Cannot annotate transition without a charge state")
```

Comment:
> `fragment_charges`  contains the "allowed" fragment charge states. 

currently this is not really checked as the charge state derived from the product can be different from the "allowed" fragment charge states. It seems we should change the code to reflect this (e.g. now it is possible that the fragment charge is equal to 1 but the allowed charges are [2,3] only).